flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer.java

```diff
@@ -248,6 +248,9 @@
     /** Flag controlling whether we are writing the Flink record's timestamp into Kafka. */
     protected boolean writeTimestampToKafka = false;
 
+    /** The transactional.id prefix to be used by the producers when communicating with Kafka. */
+    private String transactionalIdPrefix = null;
```

Comment:
nit: annotate `@Nullable` or change to `Optional`?

Reply:
Sorry for not following the Code Style and Quality Guide. I'll annotate it with `@Nullable`. Btw, since the parameter of the `setTransactionalId()` method is not annotated with `@Nullable` on #L793, which indicates it is nonnull. Shall I do a `Preconditions.checkNotNull` in the method?

Reply:
`Preconditions.checkNotNull` makes sense there :)